src/components/dialog/dialog.js

```diff
@@ -1,107 +1,293 @@
+import {keyCodes, defaultClassNames, NATIVELY_FOCUSABLE_ELEMENTS} from '../../constants';
+
+import createEl from '../../utils/createEl';
+import defer from '../../utils/defer';
+import qa from '../../utils/qa';
+
+
 /**
  * @function UIDialog
  * @version 0.0.2
+ * @desc Main UIDialog function. Creates instances of the dialog based on
+ * parameters passed in.
+ * @param {object} settings
  */
-
-import {keyCodes, defaultClassNames} from '../../constants';
-
-
 const UIDialog = ({
         dialog = '.js-dialog',
         openBtn = '.js-dialog-btn',
         closeBtn = '.js-dialog-close-btn',
         isModal = false,
+        isAlert = false,
+        readyClass = defaultClassNames.IS_READY,
+        activeClass = defaultClassNames.IS_ACTIVE,
         showBackdrop = true
     } = {}) => {
 
-    let DOM,
-        state = {};
+    // Stores all the dom nodes for the module
```

Comment:
Nit pick thing but we sometimes say "module" but it feels we're mainly calling things "components" 🐜